crates/orderbook/src/ipfs.rs

```diff
@@ -12,10 +12,13 @@ pub struct Ipfs {
 }
 
 impl Ipfs {
-    pub fn new(client: Client, base: Url, query: Option<String>) -> Self {
+    pub fn new(base: Url, query: Option<String>) -> Self {
         assert!(!base.cannot_be_a_base());
         Self {
-            client,
+            client: ClientBuilder::new()
+                .timeout(Duration::from_secs(5))
+                .build()
+                .unwrap(),
```

Comment:
This makes the HTTP client ignore "global" configuration such as `User-Agent`. The `HttpCLientFactory` has a configure method that you can use to override `ClientBuilder` configurations:

https://github.com/cowprotocol/services/blob/d04c14b87c33335da9ba70e1b7ef947b3dd13d26/crates/shared/src/http_client.rs#L36-L39

And an example of where this is used for overriding HTTP client timeouts:

https://github.com/cowprotocol/services/blob/d04c14b87c33335da9ba70e1b7ef947b3dd13d26/crates/shared/src/bad_token/token_owner_finder.rs#L245

Reply:
This code needs a custom time out which is why I intentionally did not use the global factory. I forgot about other things like user agent though. Will switch it back to the factory.